R/derive_params_growth_age.R

```diff
@@ -0,0 +1,235 @@
+#'
+#'   Expected Values: `M`, `F`
+#'
+#' @param age Current Age
+#'
+#'   A numeric vector is expected.
+#'
+#' @param age_unit Age Unit
+#
+#'   A character vector is expected.
+#'
+#'   Expected values: 'days' 'months'
+#'
+#' @param meta_criteria Metadata dataset
+#'
+#'   A metadata dataset with the following expected variables:
+#'   `AGE`, `AGEU`, `SEX`, `L`, `M`, `S`
+#'
+#'   The dataset can be derived from CDC/WHO or user-defined datasets.
+#'   The CDC/WHO growth chart metadata datasets are available in the package and will
+#'   require small modifications.
+#'   * `AGE` - Age
+#'   * `AGEU` - Age Unit
+#'   * `SEX` - Sex
+#'   * `L` - Power in the Box-Cox transformation to normality
+#'   * `M` - Median
+#'   * `S` - Coefficient of variation
+#'
+#' @param parameter Desired parameter
+#'
+#'   A condition is expected with the input dataset `VSTESTCD`/`PARAMCD`
+#'   for which we want growth derivations:
+#'
+#'   e.g. `parameter = VSTESTCD == "WEIGHT"`.
+#'
+#'   There is CDC/WHO metadata available for Height, Weight, BMI, and Head Circumference
+#'
+#' @param set_values_to_sds Variables to be set for Z-Scores
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_sds(exprs(PARAMCD = “BMIASDS”, PARAM = “BMI-for-age z-score”))`
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pairs
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @param set_values_to_pctl Variables to be set for Percentile
+#'
+#'  The specified variables are set to the specified values for the new
+#'  observations. For example,
+#'   `set_values_to_pctl(exprs(PARAMCD = “BMIAPCTL”, PARAM = “BMI-for-age percentile”))`
+#'  defines the parameter code and parameter.
+#'
+#' *Permitted Values*: List of variable-value pair
+#'
+#'  If left as default value, `NULL`, then parameter not derived in output dataset
+#'
+#' @return The input dataset additional records with the new parameter added.
+#'
+#'
+#' @family der_growth
+#'
+#' @keywords der_growth
+#'
+#' @export
+#'
+#' @examples
+#' library(dplyr)
+#' library(lubridate)
+#' library(rlang)
+#' library(admiral)
+#'
+#' advs <- dm_peds %>%
+#'   select(USUBJID, BRTHDTC, SEX) %>%
+#'   right_join(., vs_peds, by = "USUBJID") %>%
+#'   filter(USUBJID != "PEDS-1010") %>%
+#'   mutate(
+#'     VSDT = ymd(VSDTC),
+#'     BRTHDT = ymd(BRTHDTC)
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_D,
+#'     new_var_unit = CURU_D,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "days",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   derive_vars_duration(
+#'     new_var = AGECUR_M,
+#'     new_var_unit = CURU_M,
+#'     start_date = BRTHDT,
+#'     end_date = VSDT,
+#'     out_unit = "months",
+#'     trunc_out = FALSE
+#'   ) %>%
+#'   mutate(
+#'     AGECUR = ifelse(AGECUR_D >= 365.25 * 2, AGECUR_M, AGECUR_D),
```

Comment:
obviously here we have the 2 year old cut off to use whichever is the appropriate current age/unit, but depending on study this can be modfied as well, the function is agnostic to this